### PR TITLE
update plot_forecasts() function

### LINKED ISSUE #40: create facet-ready plot_forecasts() function
specs via google doc design.

#' @param forecast_data data frame with truth and forecasts from load_forecasts()
#' @param truth_data optional data frame with forecasts in the format returned 
#' by load_truth().
#' @param models model_abbr specifying model to plot. Optional if there is only
#' one model available in forecast data.
#' @param target_variable string specifying target type. It should be one of 
#' "cum death", "inc case", "inc death"
#' @param locations string for fips code or 'US'. Optional if there is only one
#' location available in forecast data.
#' @param facet interpretable facet option for ggplot
#’ @param facet_scales argument for scales in ggplot2::facet_wrap
#’ @param forecast_dates forecast dates to plot, defaults to forecast_dates present
#’ in the data
#' @param intervals values indicating which central prediction interval levels 
#' to plot, defaults to c(.5, .8, .95). NULL means only plotting point forecasts.
#' If not provided, it will default to all available intervals in forecast data.
#' @param horizon forecasts are plotted for the horizon time steps after the 
#' forecast date. Default to all available horizons in forecast data. 
#' @param truth_source character specifying where the truth data will
#' be loaded from if truth_data is not provided. 
#' Otherwise, this character specifies the data source to plot. 
#' Currently support "JHU","USAFacts" and "NYTimes".
#' @param truth_as_of the plot includes the truth data that would have been 
#' in real time as of the truth_as_of date (not using this parameter when truth data 
#' is from github repo)
#' @param title optional text for the title of the plot. if left as "default",
#' the title will be automatically generated. if "none", no title will be plotted. 
#' @param show.caption logical, if TRUE, caption will be included showing data sources
#’ @param plot logical; if TRUE, the plot is printed
#' 
#' @return invisible ggplot object
#' 
#' @export
plot_forecasts <- function(forecast_data,
                          truth_data = NULL, 
                          models,
                          target_variable,
                          locations,
                          forecast_dates,
                          facet,
                          intervals = c(.5, .8, .95),
                          horizon,
                          truth_source = "JHU",
                          truth_as_of = NULL, 
                          title = "default", 
                          show.caption = TRUE,
                          plot = TRUE)

### PR DESCRIPTION
**Pull request by Serena-Wang**
close #40

### CODE REVIEW

File: R/plot_forecast.R [hunk 1]
```diff
@@ -3,12 +3,16 @@
 #' @param forecast_data data frame with truth and forecasts from load_forecasts()
 #' @param truth_data optional data frame with forecasts in the format returned 
 #' by load_truth().
-#' @param model model_abbr specifying model to plot. Optional if there is only
+#' @param models model_abbr specifying models to plot. Optional if there is only
```
Comment (elray1) [inline R/plot_forecast.R:6]:
```suggestion
#' @param models vector of strings specifying models to plot. Optional if there is only
```

File: R/plot_forecast.R [hunk 2]
```diff
@@ -50,25 +61,25 @@ plot_forecast <- function(forecast_data,
...
   } else {
-    if (!missing(model)){
-      if (!(model %in% forecast_data$model)) {
-        stop("Error in plot_forecast: model is not available in forecast data.")
+    if (!missing(models)){
+      if (!all(models %in% forecast_data$model)) {
+        stop("Error in plot_forecast: Not all models are available in forecast data.")
       }
     } else {
       stop("Error in plot_forecast: Please select a model to plot.")
```
Comment (elray1) [inline R/plot_forecast.R:73]: Could we default to plotting all models?

File: R/plot_forecast.R [hunk 3]
```diff
@@ -50,25 +61,25 @@ plot_forecast <- function(forecast_data,
...
   } else {
-    if (!missing(location)){
-      if (!(location %in% forecast_data$location)) {
-        stop("Error in plot_forecast: location is not available in forecast data.")
+    if (!missing(locations)){
+      if (!all(locations %in% forecast_data$location)) {
+        stop("Error in plot_forecast: Not all locations are available in forecast data.")
       }
     } else {
       stop("Error in plot_forecast: Please select a location to plot.")
```
Comment (elray1) [inline R/plot_forecast.R:85]: Could we default to plotting all locations?

File: R/plot_forecast.R [hunk 4]
```diff
@@ -138,54 +160,103 @@ plot_forecast <- function(forecast_data,
...
-  } else {
-    blues <- RColorBrewer::brewer.pal(n=4, "Blues")
+    forecast_colors <- rep(tail(blues,1), length(unique(models)))
+    interval_colors <- rep(blues[1:(length(blues)-1)],
+                           length(unique(models)))
+    ribbon_colors <- blues[1:(length(blues)-1)]
   }
   
   # include truth from remote git hub repo by default
   # not using truth_as_of if we are loading truth from git hub repos
```
Comment (elray1) [inline R/plot_forecast.R:234]: Maybe we should throw a warning saying it's not being used if `truth_as_of` is provided in this case

**Review by elray1 — CHANGES_REQUESTED** [review]
A few minor requested changes here.  Also, should we increment the package version (maybe 0.1.1 or something?)

**Review by nickreich — CHANGES_REQUESTED** [review]
This review is only based on looking at the code, not running anything (time constrained today). 

Please update NEWS.md with changes in this version. should explicitly acknowledge the lack of backwards compatibility given the changing arguments in plot_forecast(). It's ok to not have the backwards compatibility, but just should say that it is there.

The vignette looks great! Looking forward to using this.

File: R/plot_forecast.R [hunk 5]
```diff
@@ -61,35 +61,19 @@ plot_forecast <- function(forecast_data,
...
   }
   
-  # validate truth_source
-  truth_source <- match.arg(truth_source, 
-                            choices = c("JHU","USAFacts", "NYTimes"), 
-                            several.ok = FALSE)
+  # optional locations parameter. Default to all locations in forecast_data
+  if (missing(locations)){
+    locations <- unique(forecast_data$location)
+  }
```
Comment (elray1) [inline R/plot_forecast.R:76]: There used to be some validation code like the below.  Do we still have something like that?  (If not, is that ok?)

```
    if (!missing(locations)){
      if (!all(locations %in% forecast_data$location)) {
        stop("Error in plot_forecast: Not all locations are available in forecast data.")
      }
    }
```

**Review by elray1 — CHANGES_REQUESTED** [review]
Not sure whether we need a change here; if not, feel free to ignore.

File: R/plot_forecast.R (hunk 5 above)
Comment (Serena-Wang) [reply R/plot_forecast.R:76]: Yeah location validation is at line 78-94.

**Review by nickreich — APPROVED** [review]
Thanks for the NEWS updates! Seems like there are maybe a few other updates to some of the additional load_latest_forecast() functions as well so that they load in location info? Might be worth documenting those as well here, but I am approving the merging of these changes all the same.

**Review by elray1 — APPROVED** [review]

**Pull request merged**